test/e2e/auto_rotation_test.go

```diff
@@ -0,0 +1,468 @@
+		out, err := exec.KubectlExec(kubeconfigPath, p.Name, p.Namespace, strings.Split(cmd, " "))
+		Expect(err).To(BeNil())
+		Expect(out).To(Equal("secret"))
+
+		// check K8s secret is valid
+		k8sSecret := secret.Get(secret.GetInput{
+			Getter:    kubeClient,
+			Name:      "rotationsecret",
+			Namespace: ns.Name,
+		})
+
+		Expect(k8sSecret).NotTo(BeNil())
+		Expect(k8sSecret.Data).NotTo(BeNil())
+		Expect(string(k8sSecret.Data["foo"])).To(Equal("secret"))
+
+		// rotate secret in key vault
+		err = kvClient.SetSecret(secretName, "rotated")
+		Expect(err).To(BeNil())
+
+		Eventually(func() bool {
+			// wait for secret to be updated
+			cmd = getPodExecCommand(fmt.Sprintf("cat /mnt/secrets-store/%s", secretName))
+			out, err = exec.KubectlExec(kubeconfigPath, p.Name, p.Namespace, strings.Split(cmd, " "))
+			Expect(err).To(BeNil())
+
+			// check K8s secret is valid
+			k8sSecret := secret.Get(secret.GetInput{
+				Getter:    kubeClient,
+				Name:      "rotationsecret",
+				Namespace: ns.Name,
+			})
+
+			Expect(k8sSecret).NotTo(BeNil())
+			Expect(k8sSecret.Data).NotTo(BeNil())
+
+			return out == string(k8sSecret.Data["foo"]) && out == "rotated"
+		}, 2*time.Minute, 15*time.Second).Should(BeTrue())
+	})
+
+	It("should auto rotate mount contents with user managed identity", func() {
+		if config.IsKindCluster {
+			Skip("test case not supported for kind cluster")
+		}
+
+		secretName := fmt.Sprintf("secret-msi-%s", utilrand.String(randomLength))
+		// create secret in keyvault
+		err := kvClient.SetSecret(secretName, "secret")
+		Expect(err).To(BeNil())
+
+		keyVaultObjects := []provider.KeyVaultObject{
+			{
+				ObjectName: secretName,
+				ObjectType: provider.VaultObjectTypeSecret,
+			},
+		}
+
+		yamlArray := provider.StringArray{Array: []string{}}
+		for _, object := range keyVaultObjects {
+			obj, err := yaml.Marshal(object)
+			Expect(err).To(BeNil())
+			yamlArray.Array = append(yamlArray.Array, string(obj))
+		}
+
+		objects, err := yaml.Marshal(yamlArray)
+		Expect(err).To(BeNil())
+
+		spc = secretproviderclass.Create(secretproviderclass.CreateInput{
+			Creator:   kubeClient,
+			Config:    config,
+			Name:      "azure",
+			Namespace: ns.Name,
+			Spec: v1alpha1.SecretProviderClassSpec{
+				Provider: "azure",
+				SecretObjects: []*v1alpha1.SecretObject{
+					{
+						SecretName: "rotationsecret",
+						Type:       string(corev1.SecretTypeOpaque),
+						Labels:     map[string]string{"environment": "test"},
+						Data: []*v1alpha1.SecretObjectData{
+							{
+								ObjectName: secretName,
+								Key:        "foo",
+							},
+						},
+					},
+				},
+				Parameters: map[string]string{
+					"keyvaultName":           config.KeyvaultName,
+					"tenantId":               config.AzureTenantID,
+					"objects":                string(objects),
+					"usePodIdentity":         "false",
+					"useVMManagedIdentity":   "true",
+					"userAssignedIdentityID": config.UserAssignedIdentityID,
```

Comment:
Where do you grant permissions to the kv resources for this identity?

Reply:
Permissions and role assignments aren't done as part of the test suite. They should be done out of band as part of e2e setup.

Reply:
In a followup PR, can you pls document prerequisite steps (or a script) for how someone can setup the e2e?

Reply:
sounds good! this is the issue https://github.com/Azure/secrets-store-csi-driver-provider-azure/issues/151 to track the script and docs.